builder/package.ubuntu-2404

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+if [[ ! -d /tmp/output/${OS_IDENTIFIER} ]]; then
+  mkdir -p "/tmp/output/${OS_IDENTIFIER}"
+fi
+
+# R 3.x requires PCRE1. On Ubuntu 24, R 3.x also requires PCRE2 for Pango support.
+pcre_libs='- libpcre2-dev'
+if [[ "${R_VERSION}" =~ ^3 ]]; then
+  pcre_libs='- libpcre2-dev
+- libpcre3-dev'
+fi
+
+deflate_libs='# - libdeflate-dev'
+if grep -q '^LIBS *=.*[-]ldeflate' ${R_INSTALL_PATH}/lib/R/etc/Makeconf; then
+    deflate_libs='- libdeflate-dev'
+fi
+
+cat <<EOF > /tmp/nfpm.yml
+name: r-${R_VERSION}
+version: 1
+version_schema: none
+section: universe/math
+priority: optional
+arch: $(dpkg --print-architecture)
+maintainer: Posit Software, PBC <https://github.com/rstudio/r-builds>
+description: |
+  GNU R statistical computation and graphics system
+vendor: Posit Software, PBC
+homepage: https://www.r-project.org
+license: GPL-2
+deb:
+  fields:
+    Bugs: https://github.com/rstudio/r-builds/issues
+depends:
+- g++
+- gcc
+- gfortran
+- libbz2-dev
+- libc6
+- libcairo2
+- libcurl4t64
+${deflate_libs}
+- libglib2.0-0t64
+- libgomp1
+- libicu-dev
+- libjpeg8
+- liblzma-dev
+- libopenblas-dev
+- libpango-1.0-0
+- libpangocairo-1.0-0
+- libpaper-utils
+${pcre_libs}
+- libpng16-16t64
+- libreadline8t64
+- libtcl8.6
+- libtiff6
+- libtirpc-dev
```

Comment:
This was mostly identical to Ubuntu 22 besides some library name changes and the addition of `libtirpc`, which was new to me but apparently required sometimes?

> https://cran.r-project.org/doc/manuals/r-release/R-admin.html#Useful-libraries-and-programs
>
> An implementation of XDR is required, and the R sources contain one which is likely to suffice (although a system version may have higher performance). XDR is part of RPC and historically has been part of libc on a Unix-alike. (In principle man xdr_string should tell you which library is needed, but it often does not: on some OSes it is provided by libnsl.) However some builds61 of glibc omit or hide it with the intention that the TI-RPC library be used, in which case libtirpc (and its development version) should be installed, and its headers62 need to be on the C include path or under /usr/include/tirpc.

It's also needed to link against libR, so included the `-dev` package as a dependency.
```sh
$ /opt/R/4.3.3/bin/R CMD config --ldflags
-Wl,--export-dynamic -fopenmp -L/usr/local/lib -L/opt/R/4.3.3/lib/R/lib -lR -lpcre2-8 -llzma -lbz2 -lz -ltirpc -lrt -ldl -lm -licuuc -licui18n
```